lib/active_merchant/billing/gateways/braintree/braintree_common.rb

```diff
@@ -18,4 +23,12 @@ def scrub(transcript)
       gsub(%r((&?ccnumber=)\d*(&?)), '\1[FILTERED]\2').
       gsub(%r((&?cvv=)\d*(&?)), '\1[FILTERED]\2')
   end
+
+  def supports_network_tokenization?
+    response = authorize(1, ActiveMerchant::Billing::NetworkTokenizationCreditCard.test_credit_card)
+    response_code = response.params['braintree_transaction'].try(:[], 'processor_response_code')
+    raise(ActiveMerchant::InvalidResponseError, response.message) unless response_code
+
+    response.params['braintree_transaction']['processor_response_code'] != ERROR_CODES.fetch(:payment_instrument_not_supported)
```

Comment:
Why not re-use `response_code` here? Also, isn't it inconsistent to raise an error here, but in the Authorize.net one (https://github.com/activemerchant/active_merchant/pull/1766/files#diff-32b8ad7eb3755b05bb1a03b6e1f9a3cdR194), we're not?